### PR TITLE
fix: 🐛 Add create_clip

### LINKED ISSUE #135: Bug: create & automate Clip function for making clips
### Title of the Issue
- [X] **Have you named the issue title clearly and descriptively?**
- [X] **Have you searched for existing issues in the project to avoid duplicates?**  

### Steps to Reproduce

- Add create_clip function to generate clips

### Changes Expected
It will create clips that can be created & then retrieved from upload & scrap.

### Checklist
- [X] I have followed the issue title guidelines.
- [X] I have included a clear and detailed description.
- [X] I have provided necessary logs, screenshots, or other files.
- [X] I am a GSSOC contributor.
- [X] I have mentioned related issues or pull requests if any.
- [X] I have provided details about the environment where the issue was observed.

### PR DESCRIPTION
**Pull request by yashksaini-coder**
✅ closes #135

This pull request includes enhancements to the `package/clip.py` file by adding docstrings to improve code documentation and introduces a new asynchronous function in `package/create.py` to create and destroy clips with a specified lifetime.

### Documentation Improvements:

* Added docstrings to `display` function in `package/clip.py` to explain its purpose and usage.
* Added docstrings to `copy_to_clipboard` function in `package/clip.py` to describe its functionality and supported operating systems.

### New Feature:

* Introduced a new asynchronous `create` function in `package/create.py` to create and destroy clips on `cl1p.net`. The function includes nested asynchronous functions `create_clip` and `destroy_clip` and uses `asyncio` to manage the clip's lifetime.

---

[sneaky-package-135.webm](https://github.com/user-attachments/assets/1c0d573d-46e0-489c-836c-4d3658880173)

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
@jindalpriyanshu101 @ShashwatAgrawal20 @Wahid7852 @this-is-yaash review this PR, and also assign it to me

File: package/create.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if destroy_success:
+            print(f"Clip destroyed successfully after {lifetime_seconds} seconds.")
+        else:
+            print("Failed to destroy the clip.")
+    else:
+        print("Failed to create the clip.")
+
+
+
+asyncio.run(create('yash1', 'This clip will self-destruct!', 15))
```
Comment (ShashwatAgrawal20) [inline package/create.py:39]: ig this should be something like 
```suggestion
if __name__ == "__main__":
    asyncio.run(create('yash1', 'This clip will self-destruct!', 15))
```

File: package/create.py
Comment (ShashwatAgrawal20) [inline package/create.py:1]: I guess you can also remove those comments, those seems way too obvious tbh!

File: package/create.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        await asyncio.sleep(lifetime_seconds)
+        
+        # Destroy the clip
+        destroy_success = await destroy_clip()
+        if destroy_success:
+            print(f"Clip destroyed successfully after {lifetime_seconds} seconds.")
+        else:
+            print("Failed to destroy the clip.")
+    else:
+        print("Failed to create the clip.")
```
Comment (ShashwatAgrawal20) [inline package/create.py:35]: And ig you can also do some exception handling and stuff too, maybe some try/except blocks.

**Review by ShashwatAgrawal20 — CHANGES_REQUESTED** [review]
Please do the required changes.

**Comment by yashksaini-coder** [comment]
hi @ShashwatAgrawal20 i have done the changes and logging off now. Let me know the status tomorrow

File: package/clip.py [hunk 3]
```diff
@@ -4,7 +4,14 @@
...
 from datetime import datetime
 import glob
+
 def display(snippet_name, password):
+    """
+    This function displays the content of a snippet file and copies it to the clipboard.
+    It first checks if the provided password matches the current time in HHMM format.
+    If the password is valid, it searches for a file with the given snippet_name in the "stash" directory.
+    If a single file is found, it reads the file content and calls the copy_to_clipboard function to copy the content.
+    """
```
Comment (this-is-yaash) [inline package/clip.py:14]:
```suggestion
```

File: package/clip.py [hunk 4]
```diff
@@ -35,6 +42,10 @@ def display(snippet_name, password):
         raise
 
 def copy_to_clipboard(text):
+    """
+    This function copies the given text to the clipboard based on the operating system.
+    It supports Linux, Windows, and macOS.
+    """
```
Comment (this-is-yaash) [inline package/clip.py:48]:
```suggestion
```
uncomment!

**Review by this-is-yaash — COMMENTED** [review]
The implementation works well!🚀

But from what I've noticed, the duration limit is kind of odd (im not sure abt the right term).

```py
asyncio.run(create('letstestthis', 'This clip will self-destruct!', 5))
```
When I pass this args, I noticed certain behaviours:
- The site is available for only 5 seconds ✅
- It's unavailable, if I open after 5 seconds ✅ 

These are good, and implemented well.

But apart from the above mentioned:
- When I visit the link and refresh the page within 5 seconds, it resets again ❌.

In simple terms, it utilized more than one view options
- view once.
- view once within time limit 
> from my observations

I'll give you an overview of how to implement them (duration part).
In the site we have **7 view options**:
image
1. Set the **View Once** (1st option) as the default (no need to mention the arguments).
2. Use `int` to match the view options.
3. `create("link", 1)` represents the 2nd option (1 Minute), and vice versa. 

What I'm trying to say is that try to match the given option with `int`. We basically made "**View Once**" default. So 2nd option is 1 minute -> pair it with `int 1`. 3rd option is 10 minutes -> pair it with `int 2`.

This is what I thought of implementing it rather than ending the duration within seconds.

If you have better approach, please let me know.
But what I need is that, I want you implement the duration part with the 7 options we have. (we don't need to mention the duration ourselves).

**Comment by yashksaini-coder** [comment]
@this-is-yaash review this PR now

**Review by Wahid7852 — APPROVED** [review]
LGTM

**Review by this-is-yaash — COMMENTED** [review]
"Duration" logic is implemented well. However, the **texts inside the site resets** when I reload the page.

Can you find a way to fix that?

**Comment by this-is-yaash** [comment]
"@yashksaini-coder, I'm merging this PR. Although the result is somewhat broken (using **view once mode with a custom time limit**), I hope we can find a way to fix it in the future. For now, I'll merge this PR as it has accomplished what was mentioned in issue #135

**Comment by yashksaini-coder** [comment]
> "@yashksaini-coder, I'm merging this PR. Although the result is somewhat broken (using **view once mode with a custom time limit**), I hope we can find a way to fix it in the future. For now, I'll merge this PR as it has accomplished what was mentioned in issue #135 

Yeah just give me few days will figure out something

**Comment by yashksaini-coder** [comment]
@this-is-yaash bro can you add labels on this PR

**Pull request merged**